src/PotentialSimulation/PotentialSimulationSetups/PotentialSimulationSetupRBCartesian3D.jl

```diff
@@ -161,7 +161,7 @@ function PotentialSimulationSetupRB(det::SolidStateDetector{T}, grid::CartesianG
             grid_boundary_factor_z_right::T = abs((z_ext[end - 1] - z_ext_mid) / (z_ext[end] - z_ext_mid))
             grid_boundary_factor_z_left::T = abs((z_ext[2] - z_ext_mid) / (z_ext[1] - z_ext_mid))
 
-            geom_weights::NTuple{3, AbstractGeometricalAxisWeights{T}} = (gw_x, gw_y, gw_z) # Weights needed for Field Simulation loop
+            geom_weights::NTuple{3, AbstractGeometricalAxisWeights{T}} = (gw_z, gw_y, gw_x) # Weights needed for Field Simulation loop

```

Comment:
This actually not only concerns this line.

For the SOR the 3D-Arrays are transformed into 4D-Arrays due to the Red-Black (even-odd) algorithm in order to be able to calculate the potential in parallel. 

Cylindrical: `[r,ϕ,z]` -> `[z, ϕ, r, even/odd]`
Cartesian: `[x,y,z]` -> `[x, y, z, even/odd]`

The order for the cylindrical case is swapped such that the inner dimension is cartesian in order to 
reduce the amount of calculation in the inner loop. 

I just changed the order here for the `geom_weights` such that indexing them gives the proper weights for 
both coordinates: E.g. `geom_weights[3]` -> `z` (cyl) and `x` (car)

In principle, it would be the best to change the order of the 3D-Arrays of the Cylindrical coordinates:
`r,ϕ,z` -> `z, ϕ, r`. However, this requires quite some changes in the whole package and would be some unusual notation.

I could also either change this back and change the order of the cylindrical-`geom_weight` (which might actually a bit more intuitive as `geom_weights[1] -> weights of the first dimension of the 4D-Array`) and/or use dispatch.